libraries/AC_WPNav/AC_WPNav.cpp

```diff
@@ -992,6 +1018,10 @@ bool AC_WPNav::set_spline_origin_and_destination(const Vector3f& origin, const V
     _flags.segment_type = SEGMENT_SPLINE;
     _flags.new_wp_destination = true;   // flag new waypoint so we can freeze the pos controller's feed forward and smooth the transition
 
+    // initialise yaw related variables
+    _track_length_xy = safe_sqrt(sq(_destination.x - _origin.x)+sq(_destination.y - _origin.y));  // horizontal track length (used to decide if we should update yaw)
+    _flags.wp_yaw_set = false;
```

Comment:
Ah, thanks very much.  there's no need for that line so I've removed it now.
You're right that we could unset the yaw where the other conditions aren't met.  the advantage of doing it that way would be that it would release control of the vehicle yaw which is useful is some rare situations.  In particular if there's an EKF reset of the heading or the vehicle's heading is yanked around by some external disturbance.  Still, it's so rare that I don't think we need to worry about it.  I'll merge after Travis says it's ok.  txs again.